### PR TITLE
✨ Add values on edit page load

### PR DESCRIPTION
**Pull request by arassaria**
# Description

Add values on edit page load
Add update function to button onclick

## Type of change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -14,6 +14,9 @@ function App() {
       <GlobalStyle />
       <Router>
         <Switch>
+          <Route path="/tours/:id/edit">
+            <AddTour concurrentTour />
```
Comment (phlgr) [inline client/src/App.js]:
```suggestion
            <AddTour />
```
Is there still need for this?

File: client/src/pages/AddTour.js [hunk 2]
```diff
@@ -123,18 +150,39 @@ export default function AddTour() {
...
+                dest,
+                date,
+                assignment,
+                priority,
+                cargo,
+                info,
+                checkboxes,
+                weekDays,
+              });
+            }
```
Comment (phlgr) [inline client/src/pages/AddTour.js]:
```suggestion
            if (id) {
            const data =  {
                  name,
                  start,
                  dest,
                  date,
                  assignment,
                  priority,
                  cargo,
                  info,
                  checkboxes,
                  weekDays,
                }
              updateData(
                {
                  collectionName: "tasks",
                  id,
                },
               data
              );
            } else {
              addTour(data);
            }
```

File: client/src/utils/api.js [hunk 3]
```diff
@@ -46,12 +46,13 @@ export async function deleteData({ collectionName, dataName }) {
...
-  await fetch(`/api/${collectionName}?data=${dataName}`, {
+export async function updateData({ collectionName, id }, props) {
+  await fetch(`/api/${collectionName}?id=${id}`, {
     method: "PATCH",
     body: JSON.stringify(props),
     headers: {
       "Content-Type": "application/json",
     },
   });
+  console.log({ props });
```
Comment (phlgr) [inline client/src/utils/api.js]:
```suggestion

```

**Review by phlgr — CHANGES_REQUESTED** [review]

File: client/src/App.js (hunk 1 above)
Comment (arassaria) [reply client/src/App.js]: Just forgot to remove this one. 😄

File: client/src/utils/api.js (hunk 3 above)
Comment (arassaria) [reply client/src/utils/api.js]: Forgot to remove this one as well 😅

File: client/src/pages/AddTour.js [hunk 4]
```diff
@@ -32,6 +33,32 @@ export default function AddTour() {
...
+          setName(data.name);
+          setStart(data.start);
+          setDest(data.dest);
+          setDate(data.date);
+          setAssignment(data.assignment);
+          setCargo(data.cargo);
+          setPriority(data.priority);
+          setInfo(data.info);
+          setCheckboxes(data.checkboxes);
+          setWeekDays(data.weekDays);
```
Comment (lmachens) [inline client/src/pages/AddTour.js]: Consider reducing all these states to a single state if you always set all of them the same time.
```suggestion
          setTask(task);
```

File: client/src/pages/AddTour.js (hunk 2 above)
Comment (arassaria) [reply client/src/pages/AddTour.js]: Thanks, although the Version suggested is not working fully. But just had to move `const data` out of the if-statement

File: client/src/pages/AddTour.js [hunk 5]
```diff
@@ -32,6 +33,32 @@ export default function AddTour() {
 
   const concurrentTour = query.get("type") === "concurrent";
 
+  useEffect(() => {
+    if (id) {
+      const doFetch = async () => {
+        try {
+          const data = await getDataByID({
```
Comment (lmachens) [inline client/src/pages/AddTour.js]: The naming is very unspecific.
What about this?
```suggestion
          const task = await getTaskById(id);
```

File: client/src/pages/AddTour.js [hunk 6]
```diff
@@ -32,6 +33,32 @@ export default function AddTour() {
...
+          setDest(data.dest);
+          setDate(data.date);
+          setAssignment(data.assignment);
+          setCargo(data.cargo);
+          setPriority(data.priority);
+          setInfo(data.info);
+          setCheckboxes(data.checkboxes);
+          setWeekDays(data.weekDays);
+        } catch (e) {
+          console.error(e);
```
Comment (lmachens) [inline client/src/pages/AddTour.js:55]: Consider writing the error in a state and render it in the UI. Otherwise, users won't notice :).

**Review by lmachens — COMMENTED** [review]
gj 👍,
but consider reworking the naming of some of your variables.

File: client/src/pages/AddTour.js (hunk 5 above)
Comment (domko95) [reply client/src/pages/AddTour.js:40]: Was trying to keep the naming that unspecific so i can keep the functions multifunctional and don't have to write new functions everytime. That way i can get Tasks, Riders and Companies from the according collection with one and the same function. Too lazy to write more functions than needed. 😄

File: client/src/pages/AddTour.js (hunk 4 above)
Comment (arassaria) [reply client/src/pages/AddTour.js]: Changes will be done tomorrow. But thanks for the advice 😄 

Opened #115 for that

File: client/src/pages/AddTour.js (hunk 6 above)
Comment (arassaria) [reply client/src/pages/AddTour.js:55]: Changes will be dealt with in #116.

**Review by bwnstck — APPROVED** [review]
lgtm ⭐

**Pull request merged**